File: src/main/java/com/guilhermebury/agalaxyfaraway/view/CharacterFragment.java
package com.guilhermebury.agalaxyfaraway.view;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.FrameLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.guilhermebury.agalaxyfaraway.R;
import com.guilhermebury.agalaxyfaraway.model.RetrofitService;
import com.guilhermebury.agalaxyfaraway.model.SWApiService;
import com.guilhermebury.agalaxyfaraway.domain.Character;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Created by Guilherme Bury on 20/04/2017.
 */

public class CharacterFragment extends Fragment {

    public CharacterFragment() {}

    private FrameLayout fragmentContainer;
    TextView title;
    ProgressDialog progress;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceStace) {
        View view = inflater.inflate(R.layout.fragment_character, container, false);

        fragmentContainer = (FrameLayout) view.findViewById(R.id.fragment_character_container);
        title = (TextView) view.findViewById(R.id.film_title);

        apiCall();

        return view;
    }

    private void apiCall() {
        progress = new ProgressDialog(getActivity());
        progress.setTitle("Searching...");
        progress.show();

        String titleString = "1";//title.getText().toString();

        retrofitConverter(titleString);
    }


    private void retrofitConverter(String titleString) {
        RetrofitService service  = SWApiService.createService(RetrofitService.class);
        Call<Character> call = service.filmCall(titleString);

        call.enqueue(new Callback<Character>() {
            @Override
            public void onResponse(Call<Character> call, Response<Character> response) {
                if (response.isSuccessful()) {
                    Character characterResponse = response.body();

                    if (characterResponse != null) {
                        title.setText(characterResponse.getTitle());

                        progress.dismiss();
                    }
                }
             }

            @Override
            public void onFailure(Call<Character> call, Throwable t) {
                Toast.makeText(getContext(), "Fail", Toast.LENGTH_SHORT);
            }
        });
    }

    public void willBeDisplayed() {
        if(fragmentContainer != null) {
            Animation fadeIn = AnimationUtils.loadAnimation(getActivity(), android.R.anim.fade_in);
            fragmentContainer.startAnimation(fadeIn);
        }
    }
    public void willBeHidden() {
        if (fragmentContainer != null) {
            Animation fadeOut = AnimationUtils.loadAnimation(getActivity(), android.R.anim.fade_out);
        }
    }
}

File: src/main/java/com/guilhermebury/agalaxyfaraway/view/CharacterFragment.kt
package com.guilhermebury.agalaxyfaraway.view

import android.app.ProgressDialog
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import android.widget.Toast

import com.guilhermebury.agalaxyfaraway.R
import com.guilhermebury.agalaxyfaraway.contract.Contract
import com.guilhermebury.agalaxyfaraway.domain.Character
import com.guilhermebury.agalaxyfaraway.presenter.PresenterImpl
import kotlinx.android.synthetic.main.fragment_character.*
import java.lang.String.valueOf

/**
 * Created by Guilherme Bury on 02/04/2019.
 */

class CharacterFragment : Fragment(), Contract.View {

    private lateinit var presenter: Contract.Presenter
    private lateinit var progress: ProgressDialog

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceStace: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_character, container, false)

        presenter = PresenterImpl(this)

        apiCall()

        return view
    }

    private fun apiCall() {

        val filmPosition = activity?.intent?.extras?.get("FILM_POSITION")
        presenter.retrofitConverter(valueOf(filmPosition))
    }

    fun willBeDisplayed() {
        if (fragmentContainer != null) {
            val fadeIn = AnimationUtils.loadAnimation(activity, android.R.anim.fade_in)
            fragmentContainer!!.startAnimation(fadeIn)
        }
    }

    fun willBeHidden() {
        if (fragmentContainer != null) {
            val fadeOut = AnimationUtils.loadAnimation(activity, android.R.anim.fade_out)
            fragmentContainer!!.startAnimation(fadeOut)
        }
    }


    override fun showLoading() {
        progress = ProgressDialog(activity)
        progress.setTitle("Searching...")
        progress.show()
    }

    override fun hideLoading() {
        progress.dismiss()
    }

    override fun showError() {
        Toast.makeText(context, "Fail", Toast.LENGTH_SHORT)
    }

    override fun onAPIInfoReceived(character: Character) {
        filmTitle.text = character.title
    }
}